docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -52,52 +52,130 @@ const theme = createTheme({
 
 {{"demo": "GlobalThemeOverride.js"}}
 
-Each component is composed of several different parts.
-These parts correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
-You can use these classes inside the `styleOverrides` key to modify the corresponding parts of the component.
+Each component is composed of several different slots.
+These slots correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
+You can use these classes inside the `styleOverrides` key to modify the corresponding slots of the component.
+
+### Overrides based on props
+
+Each slot of the component's `styleOverrides` supports a `variants` key. Variants are sets of overrides applied to the slot when the specified props match.
+
+The definitions are specified in an array. Make sure the styles that should take precedence are listed last.
```

Comment:
```suggestion
Most components support props that exist to change their design.
For example, the Card component supports a `variant` prop where you can pick `outlined` as a value that adds a border.

If you want to override a value from a specific prop, target it via the `styleOverrides` key in the theme.
This will add any new CSS you write on top of the default styles.

The override definition must specified in an array.
Also, ensure that any styles that should take precedence are listed last.
```

Maybe something more tutorial-y like this and less abstract? "Each slot's of the components' styleOverrides support a variants key" sounded so challenging 😅